### PR TITLE
Demo App Home Screen

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

 - Add a home screen to the Demo App displaying `PaymentMethodsFragment` with buttons to launch `CardFragment` or `PayPalFragment`
 - Add `SettingsFragment` where environment (sandbox or live) can be configured. 
 - Use JetPack Compose for UI elements, JetPack NavigationComponent for navigation between fragments, and JetPack Preference for demo app settings

**Note**: If you are able to, please pull down this branch and test out the app - we updated the Java version for the project to 11 to use some JetPack libraries and we want to make sure it's configured / runs correctly for all developers.

 ### Checklist

 - ~[ ] Added a changelog entry~

- @andrespelaezp 
- @tdchow 
- @sarahkoop

### CODE REVIEW

File: Demo/build.gradle [hunk 1]
```diff
@@ -62,6 +63,11 @@ dependencies {
     implementation deps.composeRuntimeLivedata
     implementation deps.composeFoundation
     implementation deps.composeMaterial
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (tdchow) [inline Demo/build.gradle]: Is the legacy-support library needed? Not sure if this was added for the Navigation Component.

File: Demo/build.gradle [hunk 2]
```diff
@@ -62,6 +63,11 @@ dependencies {
...
+    implementation deps.navigationFragment
+    implementation deps.navigationUI
+    implementation deps.preference
+    implementation 'androidx.preference:preference:1.1.1'
```
Comment (tdchow) [inline Demo/build.gradle]: Can we extract this dependency to the `deps` in the root build.gradle file?

File: Demo/build.gradle (hunk 1 above)
Comment (sshropshire) [reply Demo/build.gradle]: I've seen Android studio do this a few times I can never figure out why.

File: .idea/misc.xml [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+        <entry key="../../../../../layout/compose-model-1631723355678.xml" value="0.67" />
+        <entry key="../../../../../layout/compose-model-1631731131942.xml" value="0.5638888888888889" />
+        <entry key="../../../../../layout/compose-model-1631886742866.xml" value="0.1" />
+        <entry key="../../../../../layout/compose-model-1631908005804.xml" value="1.0824074074074075" />
+        <entry key="Demo/src/main/res/layout/activity_demo.xml" value="0.325" />
+        <entry key="Demo/src/main/res/layout/fragment_pay_pal.xml" value="0.3671875" />
+        <entry key="Demo/src/main/res/xml/root_preferences.xml" value="0.32382246376811596" />
+      </map>
+    </option>
+  </component>
```
Comment (sshropshire) [inline .idea/misc.xml:16]: q: what does this file do? I'm never sure what from the `.idea` folder should be kept in the repo.

**Review by sshropshire — APPROVED** [review]
🌊  +1 on the imports.

File: .idea/misc.xml (hunk 3 above)
Comment (tdchow) [reply .idea/misc.xml:16]: @sarahkoop and I were asking the same question. Jetbrains recommends keeping most files in the .idea directory in source control. https://intellij-support.jetbrains.com/hc/en-us/articles/206544839

Although I'm not exactly sure what the misc.xml file is used for. It seems that additional tags are added when resources are added to the project along with new composables.

Comment (sshropshire) [reply .idea/misc.xml:16]: Yeah it does kind of remind me of Xcode storyboards a bit. It jumped out during the review I'm fine with leaving as well.

File: Demo/build.gradle (hunk 1 above)
Comment (sarahkoop) [reply Demo/build.gradle]: Yeah not sure how this snuck in there - removed it.

File: Demo/build.gradle (hunk 2 above)
Comment (sarahkoop) [reply Demo/build.gradle]: I think this is a duplicate dependency - removed it 👍

**Review by tdchow — APPROVED** [review]
🚢 🚢 🚢

**Pull request merged**